### PR TITLE
Extend channel truncation params

### PR DESCRIPTION
**Pull request by driver-devel**
## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] Code changes are tested

## Changelog

- Support system messages for the channel truncate action
- Allow to specify custom `truncated_at` attribute

### CODE REVIEW

File: src/channel.ts [hunk 1]
```diff
@@ -608,13 +609,20 @@ export class Channel<
 
   /**
    * truncate - Removes all messages from the channel
-   * @param {boolean} [options.hard_delete] Defines if messages of the channel must be hard deleted
-   * @return {Promise<TruncateChannelAPIResponse<ChannelType, CommandType, UserType>>} The server response
+   * @param {TruncateOptions<AttachmentType, MessageType, UserType>} [options] Defines truncation options
+   * @return {Promise<TruncateChannelAPIResponse<ChannelType, CommandType, UserType, MessageType, ReactionType>>} The server response
    */
-  async truncate(options: { hard_delete?: boolean } = {}) {
+  async truncate(options?: TruncateOptions<AttachmentType, MessageType, UserType>) {
```
Comment (mahboubii) [inline src/channel.ts:615]:
```suggestion
  async truncate(options?: TruncateOptions<AttachmentType, MessageType, UserType> = {}) {
```

File: src/channel.ts [hunk 2]
```diff
@@ -608,13 +609,20 @@ export class Channel<
...
-      TruncateChannelAPIResponse<ChannelType, CommandType, UserType>
-    >(this._channelURL() + '/truncate', options);
+      TruncateChannelAPIResponse<
+        ChannelType,
+        CommandType,
+        UserType,
+        MessageType,
+        ReactionType
+      >
+    >(this._channelURL() + '/truncate', payload);
```
Comment (mahboubii) [inline src/channel.ts:625]:
```suggestion
    >(this._channelURL() + '/truncate', options);
```

File: src/types.ts [hunk 3]
```diff
@@ -2238,3 +2250,10 @@ export type TaskStatus = {
   };
   result?: UnknownType;
 };
+
+export type TruncateOptions<AttachmentType, MessageType, UserType> = {
+  hard_delete?: boolean;
+  message?: Message<AttachmentType, MessageType, UserType>;
```
Comment (mahboubii) [inline src/types.ts:2256]: let's pass in all the generics here:
   AttachmentType,
    ChannelType,
    CommandType,
    MessageType,
    ReactionType,
    UserType

**Review by mahboubii — APPROVED** [review]

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request merged**